clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/consumer/AsyncHandlingAutoCommitKafkaConsumer.java

```diff
@@ -276,6 +307,33 @@ protected void onPartitionsAssignedBlocking(final Set<io.vertx.kafka.client.comm
 
     @Override
     protected void onPartitionsRevokedBlocking(final Set<io.vertx.kafka.client.common.TopicPartition> partitionsSet) {
+        if (!partitionsSet.isEmpty() && offsetsCommitRecordCompletionTimeoutMillis > 0) {
+            UncompletedRecordsCompletionLatch latch = null;
+            synchronized (uncompletedRecordsCompletionLatchRef) {
+                final var uncompletedRecordsPartitions = getUncompletedRecordsPartitions(Helper.to(partitionsSet));
+                if (!uncompletedRecordsPartitions.isEmpty()) {
+                    log.info("init latch to wait up to {}ms for the completion of record handling concerning {}",
+                            offsetsCommitRecordCompletionTimeoutMillis,
+                            uncompletedRecordsPartitions.size() <= 10 ? uncompletedRecordsPartitions.keySet()
+                                    : (uncompletedRecordsPartitions.size() + " partitions"));
+                    latch = new UncompletedRecordsCompletionLatch(uncompletedRecordsPartitions);
+                    uncompletedRecordsCompletionLatchRef.set(latch);
+                }
+            }
+            if (latch != null) {
+                try {
+                    if (latch.await(offsetsCommitRecordCompletionTimeoutMillis, TimeUnit.MILLISECONDS)) {
+                        log.trace("latch to wait for the completion of record handling was released in time");
+                    } else {
+                        log.info("timed out waiting for record handling to finish after {}ms", offsetsCommitRecordCompletionTimeoutMillis);
```

Comment:
Yes, it will probably fail in the sense that the offset of the latest received record won't get committed (if the record future doesn't get marked as completed in the short period until the actual commit).